examples/kernel-dynamic-tile.cpp

```diff
@@ -0,0 +1,29 @@
+#include "RAJA/RAJA.hpp"
+
+int main(int argc, char* argv[])
+{
+
+//Using policy = KernelPolicy<Tile<tile_dynamic<0>, seq_exec, …>>;
+//RAJA::kernel_param<policy>(
+// make_tuple(RangeSegment(0,N)),
+//  make_tuple(32),  // param 0 is referenced by tile_dynamic
+//  [=](int i, int tile_size){
+//
+//  });
+
+  using namespace RAJA;
+
+  kernel_param<
+    KernelPolicy< 
+      statement::Tile<0, statement::tile_dynamic<0>, seq_exec,
+        statement::For<0, seq_exec, statement::Lambda<0> > 
+      >
+    >
+  >(make_tuple(RangeSegment{0,25}),
+      make_tuple(5),
+     //make_tuple(TileSize(10)), // not sure we need this, good for static_assert
```

Comment:
I like the idea of making this typed

Reply:
How were you thinking @ajkunen, like an extra parameter on tile_dynamic for the type of the index to be passed, perhaps defaulting to `TileSize`?